tests/python/relay/test_tidl.py

```diff
@@ -0,0 +1,210 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for TIDL compilation."""
+import os
+import sys
+
+import argparse
+parser = argparse.ArgumentParser(description="Testing TIDL code generation")
+parser.add_argument("--batch_size", type=int, default=1, help="batch size")
+parser.add_argument("--num_tidl_subgraphs", type=int, default=1, help="number of TIDL subgraphs")
+parser.add_argument("--force_arm_only", help="Force ARM-only execution", action='store_true', default=False)
+
+try:
+    args = parser.parse_args()
+except SystemExit:
+    quit()
+
+batch_size = args.batch_size
+num_tidl_subgraphs = args.num_tidl_subgraphs
+force_arm_only = args.force_arm_only
+
+if os.getenv("ARM_GCC_PATH") is None:

```

Comment:
This variable won't be set in the CI. Can you make this a helper function instead which is invoked by the tests?
```
def get_compiler_path()
  # return None if not set
```
In test:
```
arm_gcc = get_compiler_path()
if not arm_gcc:
  print('Skip compilation because ARM_GCC_PATH is not set')
  continue
lib.export_library(path_lib, cc=arm_gcc)
```

@zhiics Do any CI tests do cross compilation? How do they locate the compiler?

Reply:
Done. If cross compilation tool doesn't exist, it won't build the graph.